src/batch/src/error.rs

```
@@ -37,13 +41,34 @@ pub enum BatchError {
     #[error("Can't cast {0} to {1}")]
     Cast(&'static str, &'static str),
 
+    #[error("Storage error: {0}")]
+    Storage(
+        #[backtrace]
+        #[from]
+        StorageError,
+    ),
+
     #[error("Array error: {0}")]
     Array(
         #[from]
         #[backtrace]
         ArrayError,
     ),
 
+    #[error("Expr error: {0}")]
+    Expr(
+        #[from]
+        #[backtrace]
+        ExprError,
+    ),
+
+    #[error("Serialize/deserialize error: {0}")]
+    Serde(
+        #[source]
+        #[backtrace]
+        BoxedError,
+    ),
+
     #[error("Failed to send result to channel")]
     SenderError,
```

The source error is dropped. Is this intended?